File: Logic_TS/InputData.cpp
#include "InputData.h"
#include "EngineData.h"
#include "math.h"

InputData input;

void InputData::Load(InputData* inp) {
	//Constant
	k_const	= inp->k_const;
	t_const = inp->t_const;
	hu_const = inp->hu_const;
	c_hc_air_const = inp->c_hc_air_const;
	kf_const = inp->kf_const;
	r_const = inp->r_const;

	//Configuration values
	sigma_confg	= inp->sigma_confg;
	p_comp_n1_confg = inp->p_comp_n1_confg;
	p_comp_n2_confg = inp->p_comp_n2_confg;
	tg_confg = inp->tg_confg;
	eff_cmb_confg = inp->eff_cmb_confg;
	g_bleed_confg = inp->g_bleed_confg;
	g_cool_confg = inp->g_cool_confg;
	eff_mech_confg = inp->eff_mech_confg;
	sigma_cmb_confg = inp->sigma_cmb_confg;
	eff_turb_hp_confg = inp->eff_turb_hp_confg;
	eff_turb_lp_confg = inp->eff_turb_lp_confg;
	f_jn_confg = inp->f_jn_confg;
	thrust_confg = inp->thrust_confg * 1000;
	cv_confg = inp->cv_confg;
	ck_confg = inp->ck_confg;
	eff_coeff_n1_confg = inp->eff_coeff_n1_confg;
	eff_coeff_n2_confg = inp->eff_coeff_n2_confg;

	//Height
	min_height = inp->min_height;
	numb_of_height = inp->numb_of_height;
	max_height = inp->max_height;

	//Speed
	min_speed = inp->min_speed;
	numb_of_speed = inp->numb_of_speed;
	max_speed = inp->max_speed;

	//Temperature
	min_temperature	= inp->min_temperature;
	numb_of_temperature	= inp->numb_of_temperature;
	max_temperature	= inp->max_temperature;

	//FilePaths
	filePathLP = inp->filePathLP;
	filePathHP = inp->filePathHP;
	filePathResult = inp->filePathResult;
}


void InputData::Set_AirData_Step(InputData* inp) {
	float interval, step = 0.;

	//Calculating the interval, minimun and maximum value of the height
	interval = step = input.max_height / (input.numb_of_height - 1);

	eng.height[0] = input.min_height;

	for (int i = 1; i < input.numb_of_height; i++) {
		eng.height[i] = interval;
		interval += step;
	}

	//Calculating the interval, minimun and maximum value of the speed
	interval = step = input.max_speed / (input.numb_of_speed - 1);

	eng.speed[0] = input.min_speed;

	for (int i = 1; i < input.numb_of_speed; i++) {
		eng.speed[i] = interval;
		interval += step;
	}

	//Calculating the interval, minimun and maximum value of the temperature
	interval = step = (input.max_temperature + fabs(input.min_temperature)) / (input.numb_of_temperature - 1);

	eng.dt[0] = input.min_temperature;

	for (int i = 1; i < input.numb_of_temperature; i++) {
		eng.dt[i] = input.min_temperature + interval;
		interval += step;
	}

}


void InputData::Set_Coefficients(InputData* inp) {

	eng.c_1 = (input.k_const / (input.k_const - 1));	// = 3.5
	eng.c_2 = ((input.k_const - 1) / input.k_const);	// = 0.2857142857
	eng.c_3 = (input.kf_const / (input.kf_const - 1));	// = 4.0303030303

	eng.c_f = 0.85f;									// = 0.85
	eng.h_f = 0.15f;									// = 0.15

}

File: TurboJet_Interface/InputData.cs
using System.Runtime.InteropServices;

namespace TurboJet_Engine
{
    [StructLayout(LayoutKind.Sequential)]
    public class InputData
    {

        // Constants
        public float k_const;                   //1. adiabatic constant for dry air
        public float t_const;                   //2. value of the temperature of standart atmosphere [K]
        public float hu_const;                  //3. value of the fuel heat capacity [kJ/kg]
        public float c_hc_air_const;            //4. specific heat capacity of the air at constant pressure [J/kg*K]
        public float kf_const;                  //5. adiabatic constant for combustion products
        public float r_const;                   //6. gas constant of dry air [J/kg*K]

        // Engine configuration
        public float sigma_confg;               //7. coefficient of air pressure recovery at the inlet of the engine
        public float p_comp_n1_confg;           //8. value of air pressire increase ratio at the LP-compressor
        public float pk_n2_confg;               //9. value of air pressire increase ratio at the HP-compressor
        public float tg_confg;                  //10. value of gas temperature before the turbine [K]
        public float eff_cmb_confg;             //11. efficiency in the combustion chamber
        public float g_bleed_confg;             //12. value of relative air flow comsuption of the engine
        public float g_cool_confg;              //13. value of relative air flow comsuption for the turbine cooling
        public float eff_mech_confg;            //14. coefficient of power consumption for the auxiliary units drive
        public float sigma_cmb_confg;           //15. coefficient of air pressure recovery at the combustion chamber
        public float eff_turb_hp_confg;         //16. adiabatic efficiency of the HP-turbine stage
        public float eff_turb_lp_confg;         //17. adiabatic efficiency of the LP-turbine stage
        public float f_jn_confg;                //18. speed coefficinet of gas outflow from the nozzle
        public float thrust_confg;              //19. value of the engine thrust [kN]
        public float cv_confg;                  //20. axial velocity in gas air duct reference section at the inlet of the LP-compressor [m/s]
        public float ck_confg;                  //21. axial velocity in gas air duct reference section at the inlet of the HP-compressor [m/s]
        public float eff_coeff_n1_confg;        //22. polytropic efficiency of the axial LP-compressor stage
        public float eff_coeff_n2_confg;        //23. polytropic efficiency of the axial HP-compressor stage


        // AirSpeed points
        public float min_height;                //1. Minimum point of the height [km]
        public float numb_of_height;            //2. Number of points of the height [km]
        public float max_height;                //3. Maximum point of the height [km]

        public float min_speed;                 //1. Minimum point of the speed
        public float numb_of_speed;             //2. Number of points of the speed
        public float max_speed;                 //3. Maximum point of the speed

        public float min_temperature;           //1. Minimum point of the temperature
        public float numb_of_temperature;       //2. Number of points of the temperature
        public float max_temperature;           //3. Maximum point of the temperature


        // File paths
        public string filePathLP;               //1. FilePath of LP-compressor characteristics
        public string filePathHP;               //2. FilePath of HP-compressor characteristics
        public string filePathResult;           //3. FilePath for results save

    }
}